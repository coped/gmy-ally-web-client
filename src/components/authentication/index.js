import Login from "components/authentication/Login";
import LoginForm from "components/authentication/LoginForm";
import SignupForm from "components/authentication/SignupForm";

export { Login, LoginForm, SignupForm };
