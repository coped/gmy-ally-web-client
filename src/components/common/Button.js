import React from "react";

export default function Button(props) {
  const { classList = [], loading = false, children, ...rest } = props;
  const classes = loading
    ? ["button", "is-loading", ...classList]
    : ["button", ...classList];
  return (
    <button className={classes.join(" ")} {...rest}>
      {children}
    </button>
  );
}
