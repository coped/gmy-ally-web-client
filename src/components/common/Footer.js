import React from "react";
import logo from "images/logo_transparent.png";
import "assets/Footer.scss";

export default function Footer() {
  return (
    <div id="Footer">
      <footer className="footer">
        <div className="content has-text-centered">
          <p>
            Gym Partner is a work in progress! You can check out the source code
            and make contributions on{" "}
            <a href="https://github.com/coped/gym-partner-api">
              Gym Partner's github.
            </a>
          </p>
          <img src={logo} alt="Gym Partner logo" className="logo" />
        </div>
      </footer>
    </div>
  );
}
