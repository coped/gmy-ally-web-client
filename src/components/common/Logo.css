.Logo-animated {
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .Logo-animated {
    animation: Logo-breathe alternate infinite 1.5s;
    }
}

@keyframes Logo-breathe {
    from {
    transform: scale(1, 1);
    }
    to {
    transform: scale(1.2, 1.2);
    }
}