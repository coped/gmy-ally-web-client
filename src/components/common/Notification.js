import React from "react";

export default function Notification(props) {
  const { classList = [], children, type, ...rest } = props;
  const classes = ["notification", "is-medium", `is-${type}`, ...classList];
  return (
    <div className={classes.join(" ")} {...rest}>
      {children}
    </div>
  );
}
