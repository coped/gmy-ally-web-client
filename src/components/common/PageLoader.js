import React from "react";

export default function PageLoader({
  message = "Loading . . .",
  loading = false,
}) {
  const baseClasses = ["pageloader", "is-info"];
  const classes = loading ? [...baseClasses, "is-active"] : [...baseClasses];
  return (
    <div className={classes.join(" ")}>
      <span className="title">{message}</span>
    </div>
  );
}
