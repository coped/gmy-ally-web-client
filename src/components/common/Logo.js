import React from "react";
import "assets/images/Logo.css";
import gymPartnerLogo from "assets/images/logo_transparent.png";

export default function Logo(props) {
  return (
    <img
      src={gymPartnerLogo}
      className={props.isAnimated ? "Logo-animated" : ""}
      alt="logo"
    />
  );
}
