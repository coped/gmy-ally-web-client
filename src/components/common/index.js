import Notification from "components/common/Notification";
import Button from "components/common/Button"

export { Notification, Button };
