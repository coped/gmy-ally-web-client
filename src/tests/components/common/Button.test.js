import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Button } from "components/common";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {
  const props = {
    id: "this-button",
    classList: ["is-info", "another-class"],
    onClick: () => true,
  };
  const message = "I'm a button!";

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders with props without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button {...props}>{message}</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders children", () => {
    const wrapper = shallow(<Button {...props}>{message}</Button>);
    expect(wrapper.text()).toEqual(message);
  });

  it("uses classes from classList", () => {
    const wrapper = shallow(<Button {...props}></Button>);
    for (const className of props.classList) {
      expect(wrapper.hasClass(className)).toEqual(true);
    }
  });

  it("handles loading animation class", () => {
    const wrapper = shallow(<Button {...props}>{message}</Button>);
    wrapper.setProps({ loading: true });
    expect(wrapper.find("button").hasClass("is-loading")).toEqual(true);
    wrapper.setProps({ loading: false });
    expect(wrapper.find("button").hasClass("is-loading")).toEqual(false);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<Button {...props}>{message}</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
