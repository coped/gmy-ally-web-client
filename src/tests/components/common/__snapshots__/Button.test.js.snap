// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<button
  className="button is-info another-class"
  id="this-button"
  onClick={[Function]}
>
  I'm a button!
</button>
`;
